src/main/java/com/conveyal/r5/analyst/cluster/GridResultAssembler.java

```diff
@@ -0,0 +1,201 @@
+package com.conveyal.r5.analyst.cluster;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClient;
+import com.amazonaws.services.sqs.model.Message;
+import com.google.common.io.ByteStreams;
+import com.google.common.io.LittleEndianDataInputStream;
+import com.google.common.io.LittleEndianDataOutputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.RandomAccessFile;
+import java.util.Base64;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Assemble the results of GridComputer into AccessGrids.
+ *
```

Comment:
The thing with abstracting out access grids is that for reading it would preclude streaming access.